### DISCUSSION TITLE
Localization

### DISCUSSION THREAD

**Discussion by Bovive** [axewater/sharewarez#54] [Ideas]
I created a [branch](https://github.com/Bovive/sharewarez/tree/localization) for localization on my repo. It demonstrates one page with localization. It moves all text to a JSON file so that it can be localized to different languages.

**Things of Note**:
- Localization files are located in "/localization/{language}/{page}.json" (ex: /localization/en/discover.json)
- JSON files are per page. Alternatively, there could be one huge one. I think it is nice to break it down.
- Sets up the possibility for themes to adjust the text.

**What is Needed**:
- Admin default language setting. Makes that the language default for new users.
- User language setting. Set on their profile settings.
- Other languages. This would need to be performed by those capable of translation obviously.

Let me know what you think.

**Comment by axewater** [comment]
love it ... other things first though the userbase is super low, so this requirement can wait

**Comment by Bovive** [comment]
I agree, other languages are probably not a huge priority. There are a couple things I like about implementing an English localization:

1. It allows user customization without editing the html files. If someone wants different wording, they could edit these files.
2. Standardizing it early and adding all future text using localization files saves time from having to go back and edit things later.

If the plan isn't to add actual localization in the future, I suppose a truncated system could be used just for branding (Page headings, etc). Some people may not want the pirate theme, or may wish to have their own descriptions.

**Reply by axewater** [reply]
for sure the whole pirate theme should be optional.
took in your PR and we will go from here

**Comment by Bovive** [comment]
Sounds good. From a customization perspective, an update would overwrite any changes someone made to the main files in **/modules/static/localization.** I wonder about having a custom directory sort of like themes where those files could go. Then program it to look for the value in the custom json file first if it exists, and if it is not available then to grab it from the static json file instead.

For example I could have customization:

**/modules/custom/localization/en/discover.json**

```
{
	"heading": "Discover Booty"
}
```

It would only change the heading "Discover Loot" to "Discover Booty" and all the other localization would be pulled from **/modules/static/localization/en/discover.json**

I guess I would just need to know where you want that custom folder to be located and what to name it.
